### ISSUE TITLE
RFE: move python module to separated repo

### ISSUE THREAD

**Issue by kloczek** [dankamongmen/notcurses#1960]
Main poropose of such move would be just drop cmake build which does not adds pip .egg-info metadata.

**Comment by dankamongmen** [comment]
the only integration between python and cmake i see is this:

```
option(DUMMY_PYTHON "Build dummy python module used for compile check and Clangd" OFF)
if(${DUMMY_PYTHON})
set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
add_subdirectory("./python")
endif()
```

(which i didn't even know existed; @igo95862  added it). to build python, you enter either `cffi/` (old python) or `python/` (new python), and use `setup.py` and the standard python toolchain. i'm not sure i understand exactly what you're asking; could you be a bit more clear about what you want?

**Comment by kloczek** [comment]
On using setuptools fails in my case however it may be caused by off-tree cmake build.

**Comment by igo95862** [comment]
setuptools are not connected to cmake in any way.

Can you post any commands and outputs you are using?

**Comment by kloczek** [comment]
Just please try notrmakl sequence:
- build notcurses with separation of all build files in `build/` disrectory
- cd python
- `python setup.py build`

That will faiil because setuptools does not know anthing in such case notcurses files are in `build/`

**Comment by igo95862** [comment]
Yes. But moving it to separated repository won't solve anything.

You can either run setuptools after installing notcurses or use `C_INCLUDE_PATH` and `LIBRARY_PATH` environment variables to point the C compiler to headers and libraries.

**Comment by kloczek** [comment]
Yes it will solve **all** problems because necessary headers and aother fiiles will be available as part of system resdources (instead non-deterministic build directory)

**Comment by igo95862** [comment]
> Yes it will solve all problems because necessary headers and aother fiiles will be available as part of system resdources

After you do a `make install`. It does not matter if python binds are in the same repository or not.

**Comment by kloczek** [comment]
> It does not matter if python binds are in the same repository or not.

Yes it matter because in that siruation will be in well known location.

Sutialtion beween python bindingds and notcurses is/would be simmilar like between ncurses and notcurses.

**Comment by WSLUser** [comment]
What makes python bindings special compared to rust or other bindings? Your logic makes no sense.  The comparison to ncurses and notcurses is completely inaccurate in this case. The bindings are all in known locations and all in the same repository. Moreover, those bindings don't change the fact that the rest of the project is needed. Have you even looked at how the bindings work and what they do? All the bindings do, is provide another API surface for use with the main project. Are you suggesting that an entirely python port be created for the entire library? You're welcome to do that if you want but that would be a wasted effort by this project. There are far too many other things that need to be done before considering a full port of this library to any other language. The bindings should be enough to create and build your TUI application in python instead of C/C++.
